Datagrid/ProxyQuery.php

```diff
@@ -273,6 +285,21 @@ public function entityJoin(array $associationMappings)
     }
 
     /**
+     * Sets a query hint. If the hint name is not recognized, it is silently ignored.
+     *
+     * @param string $name  the name of the hint
+     * @param mixed  $value the value of the hint
+     *
+     * @return ProxyQueryInterface
+     */
+    final public function setHint($name, $value)
```

Comment:
Do you think this method should be added to the `ProxyQueryInterface` (commented out until next major) ? Or is it too big and should it go to a separate interface? Or in no interface?

Reply:
It is ORM specific method. For example Doctrine\ODM\PHPCR\Query\Query does not have `setHint` at all. IMHO `ProxyQueryInterface` should stay generic.